### PR TITLE
when I started working on this, only me and God knew what's going on. Now, only God does.

### PR DESCRIPTION
**Pull request by arush-sal**
* Break the mega `main.go` into `printers`, `config`, and `util` packages

* Add a `Getter` function in `util package` that accepts `(namespace, clientset, kinds, channel)` as
arguments, runs as a go routine, takes care of running individual resource's
Lister function and send the ResourceList type over the channel passed.

* Add a `Printer` function in `printers package` that accepts `(resource
interface{}, resName string)` as arguments and call the respective
resource's printer function.

* Add `GetConfigOrDie` and `GetConfig` functions in `config package` that is
now used to get the Kubernetes cluster's KUBECONFIG. 
> Disclaimer: Above functions are written by Kubernetes Author's, is provided under Apache 2.0 License and is copied as it is from the respective repository.

* Add check to ensure that `resource.Items` slice is not empty before creating `tabwriter.NewWriter`

* Add vendor folder

* Run `go mod tidy`

* Update `.gitignore`

### CODE REVIEW

**Comment by arush-sal** [comment]
fix #7

**Comment by arush-sal** [comment]
@PrasadG193 @viveksinghggits we need to pprof this thing. the CPU usage just spiked like crazy. also, can we do this any better?

**Comment by arush-sal** [comment]
@ameydev njoy ma boi

**Comment by viveksinghggits** [comment]
@PrasadG193 
I will need your help review this thing.

File: pkg/util/util.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+			handleError(err, resource)
+		case "StatefulSets":
+			list, err = clientset.AppsV1().StatefulSets(namespace).List(metav1.ListOptions{})
+			handleError(err, resource)
+		default:
+			log.Error("Given kind is not supported")
+			return
+		}
+
+		c <- list
```
Comment (viveksinghggits) [inline pkg/util/util.go:106]: @arush-sal 
The only concern that I have is if we really need to send the list of channel, can it be done by just returning the list and not using routines, I would prefer that. 
I think even if don use routines you would be able to use this in another project.
But If you think using routines is better idea and have tested the things, please confirm, i will go ahead to merge this.

**Review by viveksinghggits — CHANGES_REQUESTED** [review]

File: pkg/util/util.go (hunk 1 above)
Comment (arush-sal) [reply pkg/util/util.go:106]: @viveksinghggits I cannot think of a way through which we can return individual `ResourceList` type as we are iterate over them. The only other possible solution that I can think of is returning a slice with all the lists, then iterating over them. But that would make the overall execution blocking and would unnecessarily add a few more loops. As far as testing goes, yes, I do have tested the overall code both here(in execution) and in groot(in importing.). So I am pretty confident about this.

**Pull request merged**